static/js/composebox_typeahead.js

```diff
@@ -841,6 +857,54 @@ exports.content_typeahead_selected = function (item, event) {
         // with the topic and the final **.
         const start = beginning.length - this.token.length;
         beginning = beginning.substring(0, start) + item + '** ';
+    } else if (this.completing === 'time_jump') {
+        const flatpickr_input = $("<input id='#timestamp_flatpickr'>");
+        let timeobject;
+        let timestring = beginning.substring(beginning.lastIndexOf('!time'));
+        if (timestring.startsWith('!time(') && timestring.endsWith(')')) {
+            timestring = timestring.substring(6, timestring.length - 1);
+            moment.suppressDeprecationWarnings = true;
+            try {
+                timeobject = moment(timestring).toDate();
+            } catch {
+                // do nothing. We'll just default to showing current date for the edit.
+            }
+        }
```

Comment:
This block would do nicely as a `timerender.js` function that we can just pass the partial timestring to.